src/saturn_engine/worker_manager/services/lock.py

```diff
@@ -57,9 +57,15 @@ def lock_jobs(
             )
         )
 
+    # Join definitions and filtered out by executors
```

Comment:
This won't really work. You select all jobs from the database then filter them. But the `get_unassigned_queues` returns up to `N` jobs, so it might only select the wrong job and not return any job using your executor.

We probably need to do the other way around: read the potential jobs from the `static_definitions` with our executor selector, then `get_unassigned_queues` that match one of these job.

It's ugly and inneficient, but that's the quickest and most reliable hack for now.

Reply:
What about removing this limit when we specify an executor? 

```
if len(assigned_items) < max_assigned_items:
        assigned_items.extend(
            queues_store.get_unassigned_queues(
                session=session,
                assigned_before=assignation_expiration_cutoff,
                limit=max_assigned_items - len(assigned_items) if not lock_input.executiors,
                selector=lock_input.selector,
            )
        )
```

Reply:
I implemented the limit suggestion